src/ol/geom/multipolygon.js

```diff
@@ -409,10 +409,13 @@ ol.geom.MultiPolygon.prototype.setFlatCoordinates =
  * @param {Array.<ol.geom.Polygon>} polygons Polygons.
  */
 ol.geom.MultiPolygon.prototype.setPolygons = function(polygons) {
-  var layout = ol.geom.GeometryLayout.XY;
+  var layout = this.getLayout();
   var flatCoordinates = [];
   var endss = [];
   var i, ii, ends;
+  if(!goog.isDefAndNotNull(layout)){
```

Comment:
This whole `if`-block should not be needed. `layout` will never be `undefined`, and it may be `null`.